kiosk/src/Models/GamepadManager.ts

```diff
@@ -55,6 +71,27 @@ export class GamepadManager {
         }
     }
 
+    isCoolingDown(control: GamepadControl): boolean {
+        return !!(
+            this.cooldownTimers[control] &&
+            this.cooldownTimers[control] > Date.now()
+        );
+    }
+
+    setCooldownTimer(control: GamepadControl) {
+        this.cooldownTimers[control] =
+            Date.now() + configData.GamepadCooldownMilli;
+    }
+
+    isPressedWithCooldown(control: GamepadControl, isPressed: () => boolean): boolean {
+        if (this.isCoolingDown(control)) return false;
```

Comment:
Just a clarification question. We have this check because if the control is cooling down, then the control has already been pressed, so we don't want to trigger a press again until the cool down is done?

Reply:
That is correct. It's similar to repeat delay on a keyboard. A keyboard key may be held down continuously, but generation of "key pressed" events is throttled to a max frequency.